Sources_v3/StreamChatUI/ChatChannel/ChatMessageImageGallery.swift

```diff
@@ -0,0 +1,216 @@
+            itemPreview.didTap = attachment.flatMap { image in
+                { [weak self] in
+                    self?.didTapOnAttachment?(image)
+                }
+            }
+        }
+
+        let visiblePreviewsCount = previews.filter { !$0.isHidden }.count
+        layouts.flatMap { $0 }.forEach { $0.isActive = false }
+        layouts[max(visiblePreviewsCount - 1, 0)].forEach { $0.isActive = true }
+
+        let otherImagesCount = imageAttachments.count - previews.count
+        moreImagesOverlay.isHidden = otherImagesCount <= 0
+        moreImagesOverlay.text = "+\(otherImagesCount)"
+    }
+
+    // MARK: - Private
+
+    private func createImagePreview() -> ImagePreview {
+        uiConfig
+            .messageList
+            .messageContentSubviews
+            .imageGalleryItem
+            .init()
+            .withoutAutoresizingMaskConstraints
+    }
+}
+
+extension ChatMessageImageGallery {
+    open class ImagePreview: View {
+        public var previewURL: URL? {
+            didSet { updateContentIfNeeded() }
+        }
+
+        public var didTap: (() -> Void)?
+
+        private var imageTask: ImageTask? {
+            didSet { oldValue?.cancel() }
+        }
+
+        // MARK: - Subviews
+
+        public private(set) lazy var imageView: UIImageView = {
+            let imageView = UIImageView()
+            imageView.contentMode = .scaleAspectFill
+            imageView.layer.masksToBounds = true
+            return imageView.withoutAutoresizingMaskConstraints
+        }()
+
+        public private(set) lazy var activityIndicator: UIActivityIndicatorView = {
+            let indicator = UIActivityIndicatorView()
+            indicator.hidesWhenStopped = true
+            indicator.style = .gray
+            return indicator.withoutAutoresizingMaskConstraints
+        }()
+
+        // MARK: - Overrides
+
+        override open func setUp() {
+            let tapRecognizer = UITapGestureRecognizer(target: self, action: #selector(tapHandler(_:)))
+            addGestureRecognizer(tapRecognizer)
+        }
+
+        override open func setUpLayout() {
+            embed(imageView)
+            embed(activityIndicator)
+        }
+
+        override open func updateContent() {
+            if let url = previewURL {
+                activityIndicator.startAnimating()
+                imageTask = loadImage(with: url, options: .shared, into: imageView, completion: { [weak self] _ in
+                    self?.activityIndicator.stopAnimating()
+                    self?.imageTask = nil
+                })
+            } else {
+                activityIndicator.stopAnimating()
+                imageView.image = nil
+                imageTask = nil
+            }
+        }
+
+        // MARK: - Actions
+
+        @objc open func tapHandler(_ recognizer: UITapGestureRecognizer) {
+            didTap?()
+        }
+
+        // MARK: - Init & Deinit
+
+        deinit {
+            imageTask = nil
```

Comment:
Use `imageTask?.cancel()` instead to make your intention more clear.

Reply:
fixed in c083c1ccbb637dd5d3ce3e09302a12f2203f0678